iina/JavasctiptDevTool.swift

```diff
@@ -0,0 +1,506 @@
+//
+//  JavasctiptDevTool.swift
+//  iina
+//
+//  Created by Hechen Li on 11/30/22.
+//  Copyright © 2022 lhc. All rights reserved.
+//
+
+import SwiftUI
+import JavaScriptCore
+import WebKit
+
+
+// MARK: - Menu Controller
+
+@available(macOS 12.0, *)
+extension MenuController {
+  func createMenuItem(fromPluginInstance inst: JavascriptPluginInstance, tag: Int) -> NSMenuItem {
+    let item = NSMenuItem()
+    if inst.isGlobal {
+      item.title = "\(inst.plugin.name) (Global)"
+    } else {
+      item.title = inst.plugin.name
+    }
+    item.representedObject = inst
+    item.tag = JavasctiptDevTool.JSMenuItemInstance
+    item.target = self
+    item.action = #selector(openJavascriptDevTool)
+    return item
+  }
+
+  @objc func openJavascriptDevTool(_ sender: NSMenuItem) {
+    switch (sender.tag) {
+    case JavasctiptDevTool.JSMenuItemInstance:
+      if let inst = sender.representedObject as? JavascriptPluginInstance {
```

Comment:
```suggestion
      let instance = sender.representedObject as! JavascriptPluginInstance
```

This should never fail. If it does, someone has refactored the code and this should provide feedback for them.